src/core/jwk.rs

```diff
@@ -449,82 +449,99 @@ impl
         signature_alg: &CoreJwsSigningAlgorithm,
         msg: &[u8],
     ) -> Result<Vec<u8>, SigningError> {
-        let (padding_alg, hash) = match *signature_alg {
+        match *signature_alg {
```

Comment:
The reason why I didn't keep the `match` & `return values` pattern here is because `padding_alg` is no longer an `enum` but a trait, and this would have forced us to `Box` it to return it here.